pkg/kudoctl/util/repo/repo.go

```diff
@@ -1,15 +1,123 @@
+	Version         = "v1"
+	defaultRepoName = "community"
+)
 
 // RepositoryConfiguration represents a collection of parameters for operator repository.
 type RepositoryConfiguration struct {
-	URL string `json:"url"`
+	URL  string `json:"url"`
+	Name string `json:"name"`
+}
+
+// Repositories represents the repositories.yaml file usually in the $KUDO_HOME
+type Repositories struct {
+	RepoVersion  string                     `json:"repoVersion"`
+	Context      string                     `json:"context"`
+	Repositories []*RepositoryConfiguration `json:"repositories"`
 }
 
 // Default initialized repository.
 var Default = &RepositoryConfiguration{
-	URL: "https://kudo-repository.storage.googleapis.com",
+	Name: defaultRepoName,
+	URL:  "https://kudo-repository.storage.googleapis.com",
+}
+
+// NewRepoFile creates a new repo with only defaults populated
+func NewRepoFile() *Repositories {
+	return &Repositories{
+		RepoVersion:  Version,
+		Context:      defaultRepoName,
+		Repositories: []*RepositoryConfiguration{Default},
+	}
+}
+
+// Add appends a slice of repo configs to repositories file
+func (r Repositories) Add(repo ...*RepositoryConfiguration) {
+	r.Repositories = append(r.Repositories, repo...)
+}
+
+// SetContext switches the context to another repo config in the repositories file.  errors if no repo found.
+func (r Repositories) SetContext(context string) error {
+	repo := r.GetRepo(context)
+	if repo == nil {
+		return fmt.Errorf("no repo found with name: %s", context)
+	}
+	r.Context = context
+	return nil
+}
+
+// GetRepo returns a Repo Config for a name or nil
+func (r Repositories) GetRepo(name string) *RepositoryConfiguration {
+	for _, repo := range r.Repositories {
+		if repo.Name == name {
+			return repo
+		}
+	}
+	return nil
+}
+
+// CurrentRepo provides the repo config for the current context
+func (r Repositories) CurrentRepo() *RepositoryConfiguration {
+	return r.GetRepo(r.Context)
+}
+
+// Remove removes the repo config with the provided name
+func (r Repositories) Remove(name string) bool {
+	repos := []*RepositoryConfiguration{}
+	found := false
+	for _, repo := range r.Repositories {
+		if repo.Name == name {
+			found = true
+			continue
+		}
+		repos = append(repos, repo)
+	}
+	r.Repositories = repos
+	return found
+}
+
+// LoadRepositories reads the Repositories file
+func LoadRepositories(fs afero.Fs, path string) (*Repositories, error) {
+	b, err := afero.ReadFile(fs, path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf(
+				"could not load repositories file (%s).\n"+
+					"You might need to run `kudo init` (or "+
+					"`kudo init --client-only` if kudo is "+
+					"already installed)", path)
+		}
+		return nil, err
+	}
+
+	r := &Repositories{}
+	err = yaml.Unmarshal(b, r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// WriteFile writes a repositories file to the given path.
+func (r *Repositories) WriteFile(fs afero.Fs, path string, perm os.FileMode) error {
```

Comment:
calling writefile on repositories just feels like repository is doing too much. I would really separate the local file handling somewhere else.

Reply:
5 lines of code with control over file format... seems right to me

Reply:
Hitting the pause button on this particular thread, let's talk about it tomorrow.